### DISCUSSION TITLE
Tracking not working

### DISCUSSION THREAD

**Discussion by aashrita** [simonsobs/sodetlib#392] [Issues and Debugging]
I am currently testing the SMB and SPB. I can't get the tracking to work. It finds the tones, but then tracks 0/60 channels. I probed the FR lines for continuity already. I am not sure what to do here. Thanks!

**Comment by msilvafe** [comment]
Please post plots. Are there any log messages indicating why it is cutting channels? We need more info than this to help.

**Comment by aashrita** [comment]
Ok sure. Here are screenshots from some relevant plots.

![image](https://github.com/simonsobs/sodetlib/assets/8163556/615ab5ea-99fb-4f4c-a3a4-b52e5f7dbf22)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/4e7c54ad-d621-471c-a45e-e9ff1e0e21d4)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/2c9e9de7-ac60-4a4e-b31c-84894ed1d632)

I think most of the SMB channels are bad to be honest, but it would be nice to confirm that this isn't a problem with how I have set things up. I used Jack's method outlined in LATR wiki, and I also checked device config files.

**Reply by msilvafe** [reply]
What is "Jack's method outlined in latr wiki"? Can you link to that and/or copy paste the code u ran?

**Reply by aashrita** [reply]
https://simonsobs.atlassian.net/wiki/spaces/PRO/pages/154566905/Readout+Setup+Logs

**Reply by msilvafe** [reply]
Got it thanks. What exactly is connected here? Are there 4 smbs/spbs in a single Smurf band? Can't see exactly since the find freq plot amp portion is cut off in your screenshot. Wondering if these boxes have different enough optimal flux ramp parameters so u can't run them at the same time easily

**Reply by aashrita** [reply]
Right, so I am just checking an SMB right now. I only sent a screenshot of band 0 because that's where it's doing its thing happily. 

In the SAT right now we have one SMB, one SPB, and 2 UFMs. I am trying to just see if I can practice sodetlib stuff on the SMB and SPB before I try UFM stuff.

**Comment by jlashner** [comment]
Before running tracking, can you look at some of the resonator plots directly by disabling flux ramp, and running `uxm_relock.plot_channel_resonance`?

**Reply by aashrita** [reply]
Sorry, how do I disable the flux ramp?

**Reply by msilvafe** [reply]
S.set_feedback_enable(band, 0)
S.flux_ramp_off()

**Reply by msilvafe** [reply]
I thought you said you're working in band 0?

**Reply by aashrita** [reply]
Yeah wow, sorry, I'm going nuts. Let me redo this.

**Reply by aashrita** [reply]
![image](https://github.com/simonsobs/sodetlib/assets/8163556/c3a94700-dc34-4ee9-b802-fc8abd70a1cb)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/7b2b73e9-aa4c-4f84-9c77-217deb14c022)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/42b69cf3-e268-40d7-9ab2-242d1e5d973e)

**Reply by msilvafe** [reply]
These look like you're over driving the resonators (too much power). See how around "resonance" the points are very sparse in the circle and in the abs of s21 there's a sawtooth like edge? That's a typical when you put too much power and as you sweep your resonance you jump between two solutions for the resonance condition. 

I suggest adding warm attenuation (around 6-10dB if you have non now) and/or bumping up your attenuation to ~30 and then remaking these same plots.

**Reply by aashrita** [reply]
attenuation is at 30 currently, so let me add some warm attenuation. Do you attach right at the RF output from the SMuRF?

**Reply by msilvafe** [reply]
Output of Smurf or input to Cryostat both work

**Reply by aashrita** [reply]
I am still getting similar issues. I tested with a 10dB attenuator and a 6dB attenuator and I'm getting similar issues.
Example: 
![image](https://github.com/simonsobs/sodetlib/assets/8163556/53fb84eb-f37a-4281-819a-4ba527ed566b)

**Reply by msilvafe** [reply]
This by eye looks a bit better are you comparing the same Channels?

**Reply by aashrita** [reply]
yeah. but it still is tracking 0/60. I agree it's a bit better though.

**Reply by aashrita** [reply]
Let me know what else I can do try to play with this because I'm at the edge of my knowledge at this point :)

**Reply by msilvafe** [reply]
I'm surprised that with 6-10dB of added attenuation you still seem off in power. I wonder if your tone settings are different in your configs can you get what the synthesis scale, amplitude scale were. I seem to recall in loopback we were also seeing not quite what was expected so maybe related. Can you check those settings and report here synthesis_scale should be 1 and amplitude_scale should be 12.

May be worth leaving things for the weekend and Jack or I can walk through all the tuning steps with you on zoom sometime next week.

**Reply by aashrita** [reply]
synthesis_scale is 1 and tone_power is 12. I didn't see an amplitude_scale anywhere. i actually even copied over princeton's SMB config file and ran with that just to be sure i had settings correctly. 

This is what would hold up deployment for us right now, so I really can't afford to hold off for a week on this. We have completed basically all other tests for this final cooldown. I can quote Adrian's support on the statement that this debugging is critical for deployment of SAT UHF.

**Reply by aashrita** [reply]
By the way, this is what led me to believe that the SMB channels are just bad, but perhaps I am wrong/missing something. I ran the following code to look at individual channel tracking info, and played around with fraction_full_scale as well. I've taken screenshots of a lot of channels -- most of them look bad and there are very few that look reasonable.
```
band = 0
S3.set_feedback_enable(band,0)
S3.tracking_setup(
        band, reset_rate_khz=cfg3.dev.bands[band]['flux_ramp_rate_khz'],
        fraction_full_scale=cfg3.dev.bands[band]['frac_pp'], make_plot=True,
        save_plot=False, show_plot=True, channel=S3.which_on(band),
        nsamp=2**18, lms_freq_hz=None, meas_lms_freq=True,
        feedback_start_frac=cfg3.dev.bands[band]['feedback_start_frac'],
        feedback_end_frac=cfg3.dev.bands[band]['feedback_end_frac'],
        lms_gain=cfg3.dev.bands[band]['lms_gain'],
    )
```

![image](https://github.com/simonsobs/sodetlib/assets/8163556/02f7f4d9-f96e-44c8-888b-78e0a0d73474)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/ea12e937-f307-4714-a014-4300215bf5bf)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/7def5f3e-04d2-4e3c-9b64-0686c9e85982)
![image](https://github.com/simonsobs/sodetlib/assets/8163556/d0d6c8d5-bddc-409e-ab61-d305427910a9)

**Comment by jlashner** [comment]
Hi Aashrita, in the resonator plots you posted, like the one below: 
<img width="958" alt="image" src="https://github.com/simonsobs/sodetlib/assets/4718487/b65dcd57-7b06-40d5-86a7-b3466d6e0cb4">
it looks a lot like there is a linear slope in the phase. This is likely caused by the phase delay not being set correctly, and this screws up the resonance finding algorithm. In this plot you posted: 
<img width="937" alt="image" src="https://github.com/simonsobs/sodetlib/assets/4718487/5a3a3a9c-c212-4b81-909e-7aa6b0f1b9ee">
it looks to me like there are a lot of "channels" that it finds that are not real resonances. It is hard to tell but it also looks like deriv of the phase is slightly off from zero, which is another sign that the phase delay is not correct.

The tracking plots you post above where there is no resemblance of a SQUID curve could be channels that aren't real resonators.

I would re-run `uxm_setup.setup_phase_delay` and find-freq for the bands you expect resonators. You should only really get peak detections where it looks like there's a resonance and not all of those vertical lines away from the SPB.

**Reply by jlashner** [reply]
One simple way to check is to compare the number of channels detected compared to the number you expect from the SPBs. If they are very different then there are a lot of false channels.

**Reply by aashrita** [reply]
Hi Jack, I am running setup_phase_delay already. Also, I run full_band_resp after. I can run find_freq specifically if you prefer. As linked above for Max, I am running what you had linked on slack for the LATR instructions: https://simonsobs.atlassian.net/wiki/spaces/PRO/pages/154566905/Readout+Setup+Logs

Any deviations in my code are debugging suggestions from here/Daniel but shouldn't affect the setup.

**Reply by aashrita** [reply]
To clarify - do you want me to run setup_phase_delay twice in a row in case the first round doesnt update correctly?

**Reply by msilvafe** [reply]
The phase delay step probably failed to estimate it properly in the setup fn just run the phase delay estimation and then check a single channel to see that the slope went away. Then we can go from there.

**Reply by msilvafe** [reply]
It shouldn't be necessary in general to run twice in a row always
